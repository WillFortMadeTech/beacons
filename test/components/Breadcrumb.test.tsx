import React from "react";
import { render } from "@testing-library/react";
import {
  BreadcrumbList,
  BreadcrumbListItem,
} from "../../src/components/Breadcrumb";

describe("Breadcrumb", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<BreadcrumbList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the breadcrumb with a NextJS breadcrumb list item", () => {
    const { asFragment } = render(
      <BreadcrumbList>
        <BreadcrumbListItem link="/home">Home</BreadcrumbListItem>
      </BreadcrumbList>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the breadcrumb with a non-NextJS breadcrumb list item", () => {
    const { asFragment } = render(
      <BreadcrumbList>
        <BreadcrumbListItem link="/home" nextJSLink={false}>
          Home
        </BreadcrumbListItem>
      </BreadcrumbList>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
