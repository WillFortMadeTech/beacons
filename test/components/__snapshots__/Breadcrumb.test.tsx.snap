// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render correctly 1`] = `
<DocumentFragment>
  <div
    class="govuk-breadcrumbs "
  >
    <ol
      class="govuk-breadcrumbs__list"
    />
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb should render the breadcrumb with a NextJS breadcrumb list item 1`] = `
<DocumentFragment>
  <div
    class="govuk-breadcrumbs "
  >
    <ol
      class="govuk-breadcrumbs__list"
    >
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
          href="/home"
        >
          Home
        </a>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb should render the breadcrumb with a non-NextJS breadcrumb list item 1`] = `
<DocumentFragment>
  <div
    class="govuk-breadcrumbs "
  >
    <ol
      class="govuk-breadcrumbs__list"
    >
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
          href="/home"
        >
          Home
        </a>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
